prime-router/docs/design/proposals/0024-original-message-passthrough.md

```diff
@@ -0,0 +1,168 @@
+# Proposal for Original Message Passthrough
+
+## Background
+
+In order to meet data integrity requirements, a sender has requested a feature in the Universal Pipeline to
+preserve the original unaltered message and convey it to receivers with no alterations, including preserving the
+original file name.
+
+## Assumptions
+
+This design is intended for a specific sender. Although the design can be expanded to encompass additional senders as
```

Comment:
Right now, this is also an assumption, right?:

> The sender that has agreed to send"original functionality shall NOT send batched messages. Each submitted report shall contain only one "message", otherwise a receiver may receive duplicate messages. 

That said, I feel like we shouldn't trust the sender and verify this assumption in the pipeline, meaning not only is there special logic for this feature in the translate step but also receive/convert step, right?

Reply:
Are there any other issues you can think of with any of the other pipeline steps?

Reply:
Some implementation thoughts:

We could add ANOTHER param to the topic alongside `isUniversalPipeline` and `sendOriginal` called `allowBatch` and would default to true. This new param would be used in the receive or convert step and reject the message if it breaks the rule. I think its good that this decouples this logic from send_original, that way other topics can reuse the individual pieces as needed.